dev/tools/merge-pr.sh

```diff
@@ -1,50 +1,142 @@
 #!/usr/bin/env bash
 
 set -e
+set -o pipefail
+
+API=https://api.github.com/repos/coq/coq
 
 # This script depends (at least) on git and jq.
 # It should be used like this: dev/tools/merge-pr.sh /PR number/
 
-#TODO: check arguments and show usage if relevant
+RED="\033[31m"
+BLACK="\033[30m"
+GREEN="\033[32m"
+BLUE="\033[34m"
+YELLOW="\033[33m"
+info() {
+  echo -e "${GREEN}info:${BLACK} $1 ${BLACK}"
+}
+error() {
+  echo -e "${RED}error:${BLACK} $1 ${BLACK}"
+}
+warning() {
+  echo -e "${YELLOW}warning:${BLACK} $1 ${BLACK}"
+}
 
-PR=$1
+check_util() {
+if ! which "$1" > /dev/null; then
+  error "this script requires the $1 command line utility"
+  exit 1
+fi
+}
 
-CURRENT_LOCAL_BRANCH=$(git rev-parse --abbrev-ref HEAD)
-REMOTE=$(git config --get "branch.$CURRENT_LOCAL_BRANCH.remote")
-git fetch "$REMOTE" "refs/pull/$PR/head"
+ask_confirmation() {
+  read -p "Continue anyway? [y/N] " -n 1 -r
+  echo
+  if [[ ! $REPLY =~ ^[Yy]$ ]]
+  then
+    exit 1
+  fi
+}
 
-API=https://api.github.com/repos/coq/coq
+check_util jq
+check_util curl
+check_util git
+check_util gpg
+
+# command line parsing
+
+if [ $# != 1 ]; then
+  error "usage: $0 PR-number"
+  exit 1
+fi
+
+if [[ "$1" =~ ^[1-9][0-9]*$ ]]; then
+  PR=$1
+else
+  error "$1 is not a number"
+  exit 1
+fi
+
+# Fetching PR metadata
+
+TITLE=$(curl -s "$API/pulls/$PR" | jq -r '.title')
+info "title for PR $PR is ${BLUE}$TITLE"
 
 BASE_BRANCH=$(curl -s "$API/pulls/$PR" | jq -r '.base.label')
+info "PR $PR targets branch ${BLUE}$BASE_BRANCH"
 
+CURRENT_LOCAL_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+info "you are merging in ${BLUE}$CURRENT_LOCAL_BRANCH"
+
+REMOTE=$(git config --get "branch.$CURRENT_LOCAL_BRANCH.remote")
```

Comment:
This line is flawed.

When I tried, I got two errors. First, I was on a branch with no corresponding remote. Then I moved to a branch with remote `origin` (aka my fork on github) and not `upstream` (aka the official repo on github).

Reply:
I think I had the same problem,but I forgot to fix the script ;-)

Thanks (ask for a change please, so that I don't forget again)